mitodl/mitxonline
# Adds courserun importing from edX (like sync_courserun, but moreso)

#### Pre-Flight checklist


#### What are the relevant tickets?

No ticket for this but was inspired by #1235 

#### What's this PR do?

Adds a management command, `import_courserun`, that will create course runs in MITx Online and import the relevant data from edX in one fell swoop. Optionally, it will create CMS pages for courses that don't exist. Also optionally, it will walk a program and attempt to find all the courses in edX that exist for a given run tag and set them all up at once (so if you create a chunk of courses for a new semester, you can just tell the command the program and the new semester run tag and it'll take care of the rest). 

(Added later:) It will also optionally create products for the new courseruns. 

#### How should this be manually tested?

1. Create some courses in your local devstack instance that don't have corresponding courseruns in MITx Online. Ideally, make some that are in a program and some that are outside that program so you can test individual run generation as well as whole program generation.
2. For the individual courserun, run `manage.py import_courseware --courserun <course run ID>`. The course run should be created and its information should match the course on edX.
3. For the program, run `manage.py import_courseware --program <program> --run-tag <run tag>`. It should attempt to find all the courses that are in the program on edX and, if it finds them, then attempt to create courseruns for them. Depending on how your devstack instance is set up, you may get a bunch of errors here due to courses not existing in devstack. 
4. Apply the `--live` flag and make sure the course runs are marked as live.
5. Attempt to sync a courserun for a course that doesn't have a CMS page, and specify `--create-cms-page`. The command should create a CMS page for you.
6. The command should fail gracefully if:
   7. The course run(s) already exist
   8. The course doesn't exist 
   9. The CMS page for the course already exists
   (i.e. you should be able to re-run this for a program and have it add new things but not touch things that were already extant)

In addition, run these commands specifying `--price <price>` and ensure a product is created for the courserun. It will only create products for courseruns that it's creating, so in my testing the easiest way to test this was to delete the courseruns it created in mitxonline and re-run the command.
